src/web/wpan-controller/wpan_controller.cpp

```diff
@@ -53,85 +53,111 @@
+    snprintf(path, sizeof(path), "%s/%s", WPANTUND_DBUS_PATH,
+             mIfName);
+    scannedNetwork.SetPath(path);
+    scannedNetwork.SetDestination(GetDBusInterfaceName());
+    scannedNetwork.SetIface(WPANTUND_DBUS_APIv1_INTERFACE);
+    VerifyOrExit((ret = scannedNetwork.ProcessReply()) == 0);
+
+    VerifyOrExit((mScannedNetworkCount = scannedNetwork.GetNetworksCount()) > 0,
                  ret = kWpantundStatus_NetworkNotFound);
-    memcpy(gScanedNetowrks, scanNetwork.GetNetworks(),
-           gScannedNetworkCount * sizeof(Dbus::WpanNetworkInfo));
+    memcpy(mScannedNetworks, scannedNetwork.GetNetworks(),
+           mScannedNetworkCount * sizeof(Dbus::WpanNetworkInfo));
 exit:
     return ret;
 }
 
 const WpanNetworkInfo *WPANController::GetScanNetworksInfo(void)
 {
-    return gScanedNetowrks;
+    return mScannedNetworks;
 }
 
 int WPANController::GetScanNetworksInfoCount(void)
 {
-    return gScannedNetworkCount;
+    return mScannedNetworkCount;
 }
 
-int WPANController::Start(void)
+const char *WPANController::GetDBusInterfaceName(void)
 {
-    gInterfaceName = new DBusIfname();
-    return gInterfaceName->ProcessReply();
+
+    DBusIfname dbusIfName;
+    int        ret = kWpantundStatus_Ok;
+
+    dbusIfName.SetInterfaceName(mIfName);
+    VerifyOrExit(dbusIfName.ProcessReply() == kWpantundStatus_Ok, ret = kWpantundStatus_InvalidDBusName);
+exit:
+    return ret ? NULL : dbusIfName.GetDBusName();
 }
 
 int WPANController::Leave(void)
 {
     DBusLeave leaveNetwork;
-
-    leaveNetwork.SetDestination(gInterfaceName->GetDBusName());
+    char      path[DBUS_MAXIMUM_NAME_LENGTH + 1];
+
+    leaveNetwork.SetDestination(GetDBusInterfaceName());
+    leaveNetwork.SetInterfaceName(mIfName);
+    snprintf(path, sizeof(path), "%s/%s", WPANTUND_DBUS_PATH,
+             mIfName);
+    leaveNetwork.SetPath(path);
+    leaveNetwork.SetIface(WPANTUND_DBUS_APIv1_INTERFACE);
     return leaveNetwork.ProcessReply();
 }
 
 int WPANController::Form(const char *aNetworkname, int aChannel)
 {
     DBusForm formNetwork;
     int      ret = 0;
+    char     path[DBUS_MAXIMUM_NAME_LENGTH + 1];
 
     VerifyOrExit(aNetworkname != NULL, ret = kWpantundStatus_InvalidArgument);
     formNetwork.SetNetworkName(aNetworkname);
     VerifyOrExit((aChannel <= 26 && aChannel >= 11), ret = kWpantundStatus_InvalidArgument);
     formNetwork.SetChannelMask(aChannel);
+    formNetwork.SetInterfaceName(mIfName);
     formNetwork.SetNodeType(LEADER_ROLE);
-    formNetwork.SetDestination(gInterfaceName->GetDBusName());
+    snprintf(path, sizeof(path), "%s/%s", WPAN_TUNNEL_DBUS_PATH,
+             mIfName);
+    formNetwork.SetPath(path);
+    formNetwork.SetDestination(GetDBusInterfaceName());
+    formNetwork.SetIface(WPAN_TUNNEL_DBUS_INTERFACE);
     ret = formNetwork.ProcessReply();
 exit:
     return ret;
 }
 
-int WPANController::Join(char *aNetworkname,
+int WPANController::Join(const char *aNetworkname,
                          int16_t aChannel,
```

Comment:
`channel` type inconsistent. Sometimes `int`, sometimes `int16_t`. According to the spec, channel should be `uint16_t`.